community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -876,19 +874,21 @@ public synchronized void stop()
             awaitAllTransactionsClosed();
 
             // Write new checkpoint in the log only if the kernel is healthy.
-            // We cannot throw here since we need to shutdown without exceptions.
-            if ( databaseHealth.isHealthy() )
+            // We cannot throw here since we need to shutdown without exceptions,
+            // so let's make the checkpointing part of the life, so LifeSupport can handle exceptions properly
+            life.add( new LifecycleAdapter()
```

Comment:
Yes, unfortunately doing it that way you suggest, it is very hard to guarantee it that the check pointer is "shutdown" first. This way it might not elegant but solve the mentioned problem and touches the least amount of code since I am doing this change in a maintenance branch.